libp2p/protocols/protocol.nim

```diff
@@ -19,14 +19,16 @@ const
 
 type
   LPProtoHandler* = proc (
-    conn: Connection,
-    proto: string):
-    Future[void]
-    {.gcsafe, raises: [].}
+      conn: Connection,
+      proto: string): Future[void] {.async.}
+
+  LPProtoHandler2* = proc (
```

Comment:
hm.. 

not that I necessarily disagree with this change, but this becomes an odd case which should have been solved by covariance - ie technically, because we allow _any_ exception to be raised, we should not constrain here to `[CancelledError]`..

I wonder if there's a way to solve this with some generics, ie create a `new` that takes an `new[E](handler: proc(): InternalRaisesFuture[void, E]` - if that's possible, it's probably something we should formalize in chronos somehow, to allow expressing it without referencing `Internal..`

Reply:
sure, it would still involve going through the `new` or `handler=` instead of constructing the object manually. Agree that making it more generic would also serve the role of allowing `{.async: raises.}` annotation, but for now would have to go with `Internal`. I guess that's alright to get started.